Core/include/Managers/ComponentManager.h

```diff
@@ -1,24 +1,90 @@
 #pragma once
 
-#include "../include/Component.h"
-#include "../include/Entity.h"
-#include "../include/Utils/IDUtils.h"
+#include "Component.h"
+#include "Entity.h"
+#include "ScriptList.h"
+#include "Components/TransformComponent.h"
+#include "Components/RigidBodyComponent.h"
+#include "Components/CameraComponent.h"
+#include "Components/LightComponent.h"
+#include "Components/MeshComponent.h"
+#include "Components/ScriptComponent.h"
 
-#include <map>
+#include <string>
 
-class ComponentManager
+// Get list of components from name
```

Comment:
Yes of course we could do this with 1-3 lines if the Entity is only a few levels.
However should we import models with multiple layers, or full scenes, specifying what should or shouldn't have certain components becomes difficult to maintain. Abstracting it to a parser like we did for Solarium moves the maintaining from the Scene/Model loading code to the naming within the gltf file.

Scenario: A scene with many objects placed around, some that need rigidbodies. If we hardcode it and then change/add/remove what has rigidbodies, then we need to update the code to reflect that. If we place it in the name we can update the name in the model file and leave it to the parser.

Alternatively we could setup a Scene definition file for each scene, likely some json or xml, and define it there. This allows us to include parameters that would make the naming system messy.

Reply:
Scene.h has a TODO that suggests that Scene was supposed to become a pure virtual class, and we define multiple scenes with their own version of SpawnSceneDefinition.

With this function, wouldn't we have to track every entity in every scene in a giant switch case in this function? If so wouldn't that violate the simplicity of having Scene1.cpp or Scene2.cpp having their own scene code? Now we have to put all scene information here.

We've essentially just ping ponged it back to this file to do the same thing we would have done, except now it's in a giant conditional function instead of being right where that entity was loaded and we also have to do additonal parsing.

Can you be more specific about how you plan to implement this function because it seems like we're just doing the same thing but with extra steps

Reply:
> Scene.h has a TODO that suggests that Scene was supposed to become a pure virtual class, and we define multiple scenes with their own version of SpawnSceneDefinition.

I see that now but I think the problem still remains. The issue is specifically when we load GLTF models in as an entity, or as a Scene itself and make changes to that gltf file.  

> With this function, wouldn't we have to track every entity in every scene in a giant switch case in this function? If so wouldn't that violate the simplicity of having Scene1.cpp or Scene2.cpp having their own scene code? Now we have to put all scene information here.

No we wouldn't, if anything it makes it simpler. This function should only get called when we import entities/scenes that have multiple nodes across the gltf scene graph. When SpawnSceneDefinition() loads a gltf file as an entity, it also needs to setup the components for that entity. If we don't have some kind of parser to automatically set components for each node, then we'd have to do it in SpawnSceneDefinition() directly. For now it's fine, but what about later when we import full tracks? A track will at minimum have:

- Course
- Triggers

The Course and triggers will need to have rigidbodies attached to them for however many triggers we have. Each of those triggers will need to get attached to a script to handle collisions. We don't want to update and copy paste however many triggers a track will have. We could traverse through and set them up, but we already traverse when we load the entity anyways so why not then?

With ParseComponents it can read the name of that specific node and get from it the components and parameters for that node. Ex. Trigger1_RB_{params}_SC_CourseTrigger_(params). Keep in mind that we will likely have more objects, like collidables, that may be nested within a Scene as well.

I recognize that this isn't the best form and gets messy with scripts that have other entities as targets, but I also don't think having everything shoved into SpawnSceneDefinition is great either.